app/serializers/concerns/user_status_mixin.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module UserStatusMixin
+  def self.included(klass)
+    klass.attributes :status
+  end
+
+  def include_status?
+    @options[:include_status] && SiteSetting.enable_user_status &&
+      !object.user_option.hide_profile_and_presence && object.has_status?
```

Comment:
@jjaffeux I reused your solution from https://github.com/discourse/discourse/pull/23946/files, but I rewrote it. Your version looked like this:

```ruby
      if user.association(:user_option).loaded?
        predicate = predicate && !user.user_option.hide_profile_and_presence
      end
```

I removed the if statement.

Surely the best is to be both correct and efficient here, but if we fail at something I feel it's better to have N + 1 queries somewhere than to reveal information that's supposed to be hidden (which would happen in the solution with if – if `user_option` is not loaded the `hide_profile_and_presence` will be just ignored in that case).

I made sure to address N + 1 in places where we serialize lists of users – https://github.com/discourse/discourse/pull/24300/commits/252bf00a84da133c8ec0546ff3d4e495d774a5e1.

It may a good idea to mark the `user_option` association on the `user` model as strictly loaded at some points. But at the moment that introduces a very big amount of errors, and will also break some plugins.